packages/frontend/src/components/common/modal/ChartCreateModal/SaveToSpaceForm.tsx

```diff
@@ -1,95 +1,82 @@
-                    disabled={isLoading}
-                    size="xs"
-                    variant="default"
-                    mr="auto"
-                    compact
-                    onClick={() => {
-                        setShouldCreateNewSpace(false);
-                        // @ts-ignore, mantine form is not well typed to support generic + null value setting
-                        form.setFieldValue('newSpaceName', null);
-                    }}
-                    leftIcon={<MantineIcon icon={IconArrowLeft} />}
-                >
-                    Save to existing space
-                </Button>
-            </Stack>
+            <SpaceCreationForm
+                spaceName={newSpaceName}
+                onSpaceNameChange={(value) => {
+                    setNewSpaceName(value);
+                    // @ts-ignore - form types are complex with generics
+                    form.setFieldValue('newSpaceName', value);
+                }}
+                onCancel={() => {
+                    closeCreateSpaceForm();
+                    // @ts-ignore - form types are complex with generics
+                    form.setFieldValue('newSpaceName', null);
+                }}
+                isLoading={isLoading}
+                parentSpaceName={selectedSpaceName}
+            />
         );
     }
 
     return (
         <Stack spacing="xs">
-            <Select
-                size="xs"
-                searchable
-                label="Space"
-                description="Select a space to save the chart directly to"
-                withinPortal
-                data={spaces.map((space) => ({
-                    value: space.uuid,
-                    label: space.name,
-                }))}
-                {...form.getInputProps('spaceUuid')}
-                required
-            />
             <Can
                 I="create"
                 this={subject('Space', {
                     organizationUuid: user.data?.organizationUuid,
                     projectUuid,
                 })}
             >
-                <Button
-                    disabled={isLoading}
-                    size="xs"
-                    variant="default"
-                    mr="auto"
-                    compact
-                    leftIcon={<MantineIcon icon={IconPlus} />}
-                    onClick={() => setShouldCreateNewSpace(true)}
-                >
-                    Create new space
-                </Button>
+                <SpaceSelector
+                    spaces={spaces}
+                    selectedSpaceUuid={selectedSpaceUuid}
+                    onSelectSpace={(spaceUuid) => {
+                        setSelectedSpaceUuid(spaceUuid);
+                        // @ts-ignore - form types are complex with generics
+                        form.setFieldValue('spaceUuid', spaceUuid);
+                    }}
+                    onCreateNewSpace={() => {}} // Now handled by the button in the modal footer
```

Comment:
does it mean we should remove this property?

Reply:
removed!